hue/main.py

```diff
@@ -365,29 +365,18 @@ def discover_hue_bridges(self):
 
     def _register_sensor(self, name, external_id):
         logger.debug('Registering sensor with name %s and external_id %s', name, external_id)
-        data = {
-            'external_id': external_id,
-            'source': {'type': 'plugin', 'name': Hue.name},
+        config = {
             'name': name,
-            'physical_quantity': 'temperature',
-            'unit': 'celcius',
         }
-        response = self.webinterface.set_sensor_configuration(config=json.dumps(data))
-        data = json.loads(response)
-        if data is None or not data.get('success', False):
-            logger.error('Could not register new sensor, registration failed trough API')
-            logger.error(data)
+        response = self.webinterface.sensor.register(external_id = external_id, physical_quantity = 'temperature', unit = 'celcius', config = config)
+        if not response:
```

Comment:
Depending on what you want to do with the error, maybe add an try except block since the register function call will raise when the api call will fail. There will always be a response, from the gateway.

This way you can add back the original logging statements